frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -80,39 +106,53 @@ export const useUserSettings = <T>(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [cfLoadError, cfLoaded, fallbackLocalStorage]);
 
-  React.useEffect(() => {
-    if (sync && !isRequestPending.current) {
-      if (
-        cfData &&
-        cfLoaded &&
-        seralizeData(settingsRef.current) !== cfData?.data?.[keyRef.current]
-      ) {
-        setSettings(deseralizeData(cfData?.data?.[keyRef.current]));
-      }
-    }
-  }, [cfData, cfLoaded, sync]);
-
   const callback = React.useCallback<React.Dispatch<React.SetStateAction<T>>>(
     (action: React.SetStateAction<T>) => {
-      if (isRequestPending.current) return;
       const previousSettings = settingsRef.current;
       const newState =
         typeof action === 'function' ? (action as (prevState: T) => T)(previousSettings) : action;
       setSettings(newState);
       if (cfLoaded) {
-        isRequestPending.current = true;
+        increaseRequest();
         updateConfigMap(cfData, keyRef.current, seralizeData(newState))
           .then(() => {
-            isRequestPending.current = false;
+            decreaseRequest();
           })
           .catch(() => {
+            decreaseRequest();
             setSettings(previousSettings);
-            isRequestPending.current = false;
           });
       }
     },
-    [cfData, cfLoaded],
+    [cfData, cfLoaded, decreaseRequest, increaseRequest],
   );
 
-  return fallbackLocalStorage ? [lsData, setLsDataCallback, true] : [settings, callback, loaded];
+  const resultedSettings = React.useMemo(() => {
+    /**
+     * If key is deleted from the config map then return default value
+     */
+    if (
+      cfLoaded &&
+      cfData &&
+      !cfData.data?.hasOwnProperty(keyRef.current) &&
+      settings &&
+      !isRequestPending
+    ) {
+      return defaultValueRef.current;
+    }
```

Comment:
Shouldn't this only happen if `sync == true` ?
And why does it matter if `settings` is truthy?

Reply:
We are checking for settings here because if `settings`is undefined then it means we haven't initialized the key and that should happen with the patch request in the `useEffect`. I guess we should specifically check for `settings !== undefined`.
> Shouldn't this only happen if sync == true

I'm not sure here because if user deleted key from config map that would be to reset the state and it made sense to return default value here regardless of sync.
Another question that comes to mind is should also make patch request with the default value or call `setSettings` with defaultValue.